config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+class Rack::Attack
+  class Request < ::Rack::Request
+    # Take remote IP from Cloudfare's headers instead of rev proxy IP
+    def remote_ip
+      if ENV["CF_PROXY_DISABLED"]
+        @remote_ip ||= ip
+      else
+      # Cloudflare stores remote IP in CF_CONNECTING_IP header
+        @remote_ip ||= (env["HTTP_CF_CONNECTING_IP"] ||
+                        env["action_dispatch.remote_ip"] ||
+                        ip).to_s
+      end
+    end
+
+    # Hack to get JSON request params
+    # Reads from the IO stream and resets the stream
+    def json_params
```

Comment:
Why was this needed ?

Reply:
Rack requests do not have any better way to parse JSON params

Reply:
In that case we need to handle `JSON::ParserError` as well